pkg/trace/api/proxy.go

```diff
@@ -0,0 +1,192 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/logutil"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type proxyEndpointsConfig struct {
```

Comment:
Why not `telemetryProxy`?

Reply:
This code was meant to try to be plugin replacement for other proxies in the agent right now. 
The other proxies are not that much different, its mostly superficial and I think could be unified with right abstraction:
The differences mainly are;
- URL building, and handling, some forward `prefix/` some forward full Path.
- Forwarding, sending data to a single DC or multiple via additional_endpoints - proxyEndpointsConfig if additional endpoints cfg option are not specified, it will only ever send to a single DC
- Attaching metadata, either they do not attach metadata or they do, in 2-3 different ways - this can be handled via additionalDirector thats executed on every request

In short making this proxy  work for  ProfilingProxy only requires appropriate data in proxyEndpointsConfig, and a custom additionalDirector setting correct metadata. Otherwise the code will do the same things to the request.

AppSec proxy, requires additionalDirector, and a change to disabledMessage.

Same for debugger proxy

I haven't created a PR to unify this, but at least I was hoping this change will help other teams in the future. So the code reflects that to an extent.